### ISSUE TITLE
sleigh, tips on type casting based on addressing mode

### ISSUE THREAD

**Issue by mumbel** [NationalSecurityAgency/ghidra#2235]
Working on a variable length instruction processor that uses addressing modes for any source operand(s) or destination operand of any instruction (looks like the only difference is destination wouldn't be immediate or literal modes).  One of the addressing modes modes acts as an extended mode to another addressing mode and casts that value to: `ubyte`, `sbyte`, `shalf`, `uhalf`, `uword` or `word` as a way to override the size/sign of an operand.  For example:  `MOVB {sbyte}%r0,{word}%r0`.  Which I read as `int x = (int)((char)((unsigned char)y));` I guess.

I have not figured out a good method to do the destination's cast before assignment.  The approach I thought would work ends up reading dst, casting that read value, and then assigning to that value (PCODE never writes to `r0` in my example).

Any suggestions?

**Comment by emteere** [comment]
Can you post your example?  Would like to see the constructor/subconstructors involved.
My suspicion is you are possibly using the same sub-constructor to match for both addressing mode dst and src.

**Comment by mumbel** [comment]
Yes, I am.  I could have sworn I tried splitting them up, maybe I was in a bad state.

https://github.com/mumbel/ghidra_we

https://archives.loomcom.com/3b2/scsi/  has a sample we32k binary

**Comment by astrelsky** [comment]
> Yes, I am. I could have sworn I tried splitting them up, **maybe I was in a bad state.**
> 
> https://github.com/mumbel/ghidra_we
> 
> https://archives.loomcom.com/3b2/scsi/ has a sample we32k binary

Did you try turning yourself off and back on?

Couldn't help myself.
